### PR TITLE
Add help for iocsh commands by zioven

### PR DESCRIPTION
**Pull request by zioven**
Hi,

This is a pull request for the changes to iocSh help done during the EPICS code-a-thon March 2021 done by me
(there is a [pull request](https://github.com/epics-base/epics-base/pull/136) by @oksanagit and I believe there is one incoming from @ebernalru)

Below is a list of functions implemented in this pull request (information mostly gathered from AppDevGuide and some from source code or behavioir on `iocsh`) .

Please review and comment: 

**NOTES:**
`[OK]` - happy with solution OK (copy paste from AppDevGuide)
`[OK*]` - should be OK, but please check 
`[PART]` - documented to the best of my knowledge/understanding, please review and comment

---

```
[OK*]   modules/database/src/ioc/rsrv/rsrvIocRegister.c casr

[PART]  modules/database/src/ioc/as/asIocRegister.c     asDumpHash

[PART]  modules/libcom/src/osi/osiNTPTime.c             Report
[PART]  modules/libcom/src/osi/osiNTPTime.c             Shutdown
[PART]  modules/libcom/src/osi/osiClockTime.c           Report
[PART]  modules/libcom/src/osi/osiClockTime.c           Shutdown

[OK]    modules/libcom/src/iocsh/iocsh.cpp              var
[OK]    modules/libcom/src/iocsh/iocsh.cpp              help
[OK]    modules/libcom/src/iocsh/iocsh.cpp              iocshCmd
[PART]  modules/libcom/src/iocsh/iocsh.cpp              iocshLoad
[OK]    modules/libcom/src/iocsh/iocsh.cpp              iocshRun
[OK]    modules/libcom/src/iocsh/iocsh.cpp              on
[OK]    modules/libcom/src/iocsh/iocsh.cpp              comment

[OK]    modules/libcom/src/iocsh/libComRegister.c       date
[OK]    modules/libcom/src/iocsh/libComRegister.c       echo
[OK]    modules/libcom/src/iocsh/libComRegister.c       chdir
[OK]    modules/libcom/src/iocsh/libComRegister.c       pwd
[OK]    modules/libcom/src/iocsh/libComRegister.c       epicsEnvSet
[OK]    modules/libcom/src/iocsh/libComRegister.c       epicsEnvUnset
[OK]    modules/libcom/src/iocsh/libComRegister.c       epicsParamShow
[OK]    modules/libcom/src/iocsh/libComRegister.c       epicsPrtEnvParams
[OK]    modules/libcom/src/iocsh/libComRegister.c       epicsEnvShow
[PART]  modules/libcom/src/iocsh/libComRegister.c       registryDump
[OK]    modules/libcom/src/iocsh/libComRegister.c       iocLogInit
[OK]    modules/libcom/src/iocsh/libComRegister.c       iocLogDisable
[PART]  modules/libcom/src/iocsh/libComRegister.c       iocLogShow
[OK]    modules/libcom/src/iocsh/libComRegister.c       eltc
[OK]    modules/libcom/src/iocsh/libComRegister.c       errlogInit
[OK]    modules/libcom/src/iocsh/libComRegister.c       errlogInit2
[OK]    modules/libcom/src/iocsh/libComRegister.c       errlog
[OK]    modules/libcom/src/iocsh/libComRegister.c       iocLogPrefix
[OK]    modules/libcom/src/iocsh/libComRegister.c       epicsThreadShowAll
[OK]    modules/libcom/src/iocsh/libComRegister.c       thread
[PART]  modules/libcom/src/iocsh/libComRegister.c       taskwdShow
[OK]    modules/libcom/src/iocsh/libComRegister.c       epicsMutexShowAll
[OK]    modules/libcom/src/iocsh/libComRegister.c       epicsThreadSleep
[OK]    modules/libcom/src/iocsh/libComRegister.c       epicsThreadResume
[OK]    modules/libcom/src/iocsh/libComRegister.c       generalTimeReport
[OK]    modules/libcom/src/iocsh/libComRegister.c       installLastResortEventProvider
```

### CODE REVIEW

File: modules/database/src/ioc/rsrv/rsrvIocRegister.c [hunk 1]
```diff
@@ -20,7 +20,11 @@
 /* casr */
 static const iocshArg casrArg0 = { "level",iocshArgInt};
 static const iocshArg * const casrArgs[1] = {&casrArg0};
-static const iocshFuncDef casrFuncDef = {"casr",1,casrArgs};
+static const iocshFuncDef casrFuncDef = {"casr",1,casrArgs,
+                                         "Channel Access Server Report with following levels:\n"
+                                         "  0 - server’s protocol version level and summary for each attached client\n"
+                                         "  1 - extends report with information about connected clients and beacons\n"
+                                         "  2 - extends report with specific channel names and UDP search requests\n"};
```
Comment (mdavidsaver) [inline modules/database/src/ioc/rsrv/rsrvIocRegister.c]: fyi. There are higher levels up detail, up to 10 I think.  The output is highly expert friendly.  I'm not sure how to express this.  Maybe `  3+ - expert`?  Maybe not worth mentioning?

File: modules/libcom/src/iocsh/libComRegister.c [hunk 2]
```diff
@@ -64,7 +66,8 @@ IOCSH_STATIC_FUNC void echo(char* str)
 
 static const iocshArg echoArg0 = { "string",iocshArgString};
 static const iocshArg * const echoArgs[1] = {&echoArg0};
-static const iocshFuncDef echoFuncDef = {"echo",1,echoArgs};
+static const iocshFuncDef echoFuncDef = {"echo",1,echoArgs,
+                                         "Print any string (can aslo print EPICS environment variables)\n"};
```
Comment (mdavidsaver) [inline modules/libcom/src/iocsh/libComRegister.c]: aslo -> also

Maybe: "Print string after expanding macros and environment variables".

File: modules/database/src/ioc/rsrv/rsrvIocRegister.c (hunk 1 above)
Comment (zioven) [reply modules/database/src/ioc/rsrv/rsrvIocRegister.c]: expanded in a271e4dca11964a0b899b33b8811451e2308b867

File: modules/libcom/src/iocsh/libComRegister.c (hunk 2 above)
Comment (zioven) [reply modules/libcom/src/iocsh/libComRegister.c]: adjusted to suggestion in 8711dcb7b57d0627de06b3cae10ff73dc8a84095

**Comment by mdavidsaver** [comment]
Applied as 746d21c71de52e0a2a5c09cfaf9f4ee53ced09fc.  Thanks.

**Pull request closed without merging**